src/parser/ast.rs

```diff
@@ -1004,5 +1076,153 @@ mod tests {
         for secs in cases {
             assert!(AtModifier::try_from(secs).is_err())
         }
+
+        assert_eq!(
+            AtModifier::try_from(token::T_ADD),
+            Err("invalid @ modifier preprocessor '+', START or END is valid.".into())
+        );
+
+        assert_eq!(
+            AtModifier::try_from(Expr::from("string literal")),
+            Err("invalid float value after @ modifier".into())
+        );
+    }
+
+    #[test]
+    fn test_binary_labels() {
+        assert_eq!(
+            VectorMatchModifier::On(HashSet::from([String::from("foo"), String::from("bar")]))
+                .labels(),
+            &HashSet::from([String::from("foo"), String::from("bar")])
+        );
+
+        assert_eq!(
+            VectorMatchModifier::Ignoring(HashSet::from([
+                String::from("foo"),
+                String::from("bar")
+            ]))
+            .labels(),
+            &HashSet::from([String::from("foo"), String::from("bar")])
+        );
+
+        assert_eq!(
+            VectorMatchCardinality::OneToMany(HashSet::from([
+                String::from("foo"),
+                String::from("bar")
+            ]))
+            .labels()
+            .unwrap(),
+            &HashSet::from([String::from("foo"), String::from("bar")])
+        );
+
+        assert_eq!(
+            VectorMatchCardinality::ManyToOne(HashSet::from([
+                String::from("foo"),
+                String::from("bar")
+            ]))
+            .labels()
+            .unwrap(),
+            &HashSet::from([String::from("foo"), String::from("bar")])
+        );
+
+        assert_eq!(VectorMatchCardinality::OneToOne.labels(), None);
+        assert_eq!(VectorMatchCardinality::ManyToMany.labels(), None);
+    }
+
+    #[test]
+    fn test_neg() {
+        assert_eq!(
+            -VectorSelector::from("foo"),
+            UnaryExpr {
+                expr: Box::new(Expr::from(VectorSelector::from("foo")))
+            }
+        )
+    }
+
+    #[test]
+    fn test_scalar_value() {
+        assert_eq!(Some(1.0), Expr::from(1.0).scalar_value());
+        assert_eq!(None, Expr::from("1.0").scalar_value());
+    }
+
+    #[test]
+    fn test_at_expr() {
+        let vs =
+            Expr::from(VectorSelector::from("foo")).at_expr(AtModifier::try_from(1.0).unwrap());
+        assert!(vs.is_ok());
```

Comment:
I'd like to unwrap directly, so you could get the error message if the case fails. `assert!(x.is_ok())` just tells you vs is `Err` but doesn't print the error message.
```suggestion
        let vs =
            Expr::from(VectorSelector::from("foo")).at_expr(AtModifier::try_from(1.0).unwrap()).unwrap();
```